### PR TITLE
Anonymization of exam sheets

### PR DESCRIPTION
**Pull request by Madek1**

### CODE REVIEW

File: research/exam-sheets-anonymization/main.py
Comment (Krystian030) [inline research/exam-sheets-anonymization/main.py:1]: I suggest using if __name__ == "__main__": to wrap the script's main execution code.

Comment (Krystian030) [reply research/exam-sheets-anonymization/main.py:1]: Also I recommend adding logging or print statements to the script to provide information on how many files have been processed and how many remain

Comment (Krystian030) [reply research/exam-sheets-anonymization/main.py:1]: Add README file with instructions on how to use the script

File: research/exam-sheets-anonymization/main.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from os import listdir
+from os.path import splitext
+import cv2
+import numpy as np
+import matplotlib.pyplot as plt
+import math
+from random import randint
+from pathlib import Path
+import sys
+import tqdm
```
Comment (Krystian030) [inline research/exam-sheets-anonymization/main.py]: Here are the unused imports

Comment (Krystian030) [reply research/exam-sheets-anonymization/main.py]: Also I don't see the imports for functions from utils.py

File: research/exam-sheets-anonymization/utils.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      s_m = square[0] + (square[2] / 2)
+      if m - 2 * w <= s_m <= m + 2 * w:
+        count += 1
+  return count
+
+def find_bottom_left_square(squares, image):
+  """
+  Find left bottom square
+  """
+  height, _, _ = img.shape
```
Comment (Krystian030) [inline research/exam-sheets-anonymization/utils.py]: img variable doesn't exist

File: research/exam-sheets-anonymization/utils.py
Comment (Krystian030) [inline research/exam-sheets-anonymization/utils.py:1]: Please use ctrl + alt + L to format the code and ctrl + alt + O to optimize imports before committing changes.

File: research/exam-sheets-anonymization/main.py
Comment (Krystian030) [inline research/exam-sheets-anonymization/main.py:1]: Please use ctrl + alt + L to format the code and ctrl + alt + O to optimize imports before committing changes.

**Review by Krystian030 — CHANGES_REQUESTED** [review]
I've made a few additions to the script. Please note that the script doesn't work propery for data from https://luz6.eti.pg.gda.pl:7778/s/O9X4WgtB8KkV8WE. Additionally, I encountered some initial difficulties running the script, hence the additions. Regarding memory usage, it's currently within reasonable limits (<10 GB), so the optimization is not required.

File: research/exam-sheets-anonymization/main.py
Comment (Madek1) [reply research/exam-sheets-anonymization/main.py:1]: I can format the code, but JFYI these shortcuts only work in the JetBrains IDEs ;)

File: research/exam-sheets-anonymization/utils.py
Comment (Madek1) [reply research/exam-sheets-anonymization/utils.py:1]: I can format the code, but JFYI these shortcuts only work in the JetBrains IDEs ;)

File: research/exam-sheets-anonymization/main.py
Comment (Madek1) [reply research/exam-sheets-anonymization/main.py:1]: There is also no default formatter specified for this project. So I think we should do this in a separate task

File: research/exam-sheets-anonymization/utils.py
Comment (Madek1) [reply research/exam-sheets-anonymization/utils.py:1]: There is also no default formatter specified for this project. So I think we should do this in a separate task

File: research/exam-sheets-anonymization/main.py
Comment (Madek1) [reply research/exam-sheets-anonymization/main.py:1]: I can add a readme file and improve the log system (a simple system already exists, you can search for "print" in the code)

Comment (Madek1) [reply research/exam-sheets-anonymization/main.py:1]: Fixed

**Review by Krystian030 — APPROVED** [review]
Double checked, LGTM

**Review by S-cichocka — APPROVED** [review]

**Review by Krystian030 — APPROVED** [review]

**Pull request merged**